src/backend/distributed/connection/shared_connection_stats.c

```diff
@@ -587,6 +589,99 @@ SharedConnectionStatsShmemInit(void)
 }
 
 
+/*
+ * AdaptiveConnectionManagementFlag returns the appropriate connection flag
+ * for any given activeConnectionCount to remote nodes.
+ *
+ * This function should only be called if the code-path is capable of handling
+ * optional connections.
+ */
+int
+AdaptiveConnectionManagementFlag(int activeConnectionCount)
+{
+	if (UseConnectionPerPlacement())
+	{
+		/*
+		 * User wants one connection per placement, so no throttling is desired
+		 * and we do not set any flags.
+		 *
+		 * The primary reason for this is that allowing multiple backends to use
+		 * connection per placement could lead to unresolved self deadlocks. In other
+		 * words, each backend may stuck waiting for other backends to get a slot
+		 * in the shared connection counters.
+		 */
+		return 0;
```

Comment:
@SaitTalhaNisanci noted in the previous PR if we can have a macro or such. I thought about it, but this function returns a `connectionFlag`  and  just for this reason I don't want to add  a new flag.

If we return a macro instead of flags, I think it might be confusing.

Reply:
We have a CONNECTION_PER_PLACEMENT flag but we return no flags if `UseConnectionPerPlacement` is true, isn't this confusing?

Reply:
The naming of that flag is updated in https://github.com/citusdata/citus/pull/4054/files

`CONNECTION_PER_PLACEMENT ` is not a great name

Reply:
I guess with returning 0, the question is `what does it mean to return a 0`, or in other words `what does 0 map to in this context` I find it useful to have that explicitly written in the code, instead of a comment. Why do you think it might be confusing?

Reply:
> Why do you think it might be confusing?

Because in other `if/else` parts  we return a value from `enum MultiConnectionMode`, and in this one we return a different macro/value. That's a no-go for me.

> what does it mean to return a 0

Here it means that `no adaptive connection management required`.

One idea, as I noted above, could be add one more flag to the `MultiConnectionMode`, but that sounds too much to me for this case. In the code, the convention is to pass `connectionFlags = 0` in many places. That's probably why this doesn't bother me much. 

 I guess we should consider improving all of them with a separate pr.

Reply:
> I guess we should consider improving all of them with a separate pr.

Makes sense